presto-hive/src/test/java/com/facebook/presto/hive/metastore/TestingHiveMetastore.java

```diff
@@ -354,6 +354,32 @@ public synchronized void renameColumn(String databaseName, String tableName, Str
     }
 
     @Override
+    public synchronized void dropColumn(String databaseName, String tableName, String columnName)
+    {
+        SchemaTableName name = new SchemaTableName(databaseName, tableName);
+        Table oldTable = getRequiredTable(name);
+
+        if (!oldTable.getColumn(columnName).isPresent()) {
+            throw new ColumnNotFoundException(name, columnName);
+        }
+        if (oldTable.getPartitionColumns().size() == 1) {
```

Comment:
This check seems to be wrong. It should prevent dropping any partition columns and prevent dropping the only data column.